src/libraries/System.Net.WebSockets/src/System/Net/WebSockets/Compression/WebSocketDeflater.cs

```diff
@@ -0,0 +1,191 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using static System.IO.Compression.ZLibNative;
+
+namespace System.Net.WebSockets.Compression
+{
+    /// <summary>
+    /// Provides a wrapper around the ZLib compression API.
+    /// </summary>
+    internal sealed class WebSocketDeflater : IDisposable
```

Comment:
How is it guaranteed that it will not be used concurrently?

Reply:
The guarantee comes from the ManagedWebSocket class itself. It disallows concurrent sends or receives.